### ISSUE TITLE
Guest hang when host filesystem full

### ISSUE THREAD

**Issue by wudx05** [kimchi-project/kimchi#367]
According to the qemu document[1] ,  the default error action for write is enospc.  The policy enospc will make the failure request caused by no space re-queued and vm stopped. The request will be resumed after more host storage space is available.  So the problem here is not the setting of error policy.  

What should be improved on kimchi side is the handling of the io error event.  Actually, qemu will raise a qmp event to libvirt on io error (see function bdrv_error_action()).  The management application based on libvirt can register callback to handle this kind of event. If the vm image is stored on a lv volume whose size is less than the virtual size of qcow2 image,  it can extend the lv. If the vm image is file based volume,  what the management app can do is just raising a event to front end because it needs human intervene.  Currently, kimchi doesn't have callback for the I/O error event and doesn't support thin-provisioning on LVM.  So we can do is just reporting an event to front end. Unfortunately,  kimchi doesn't have a facility of asynchronous event report.  

To sum up, kimchi needs add:
an event report facility to let UI get alert on vm events.
a callback for libvirt's IO_ERROR_REASON event and raise a warning to front end.

[1] http://qemu.weilnetz.de/qemu-doc.html
werror=action,rerror=action
Specify which action to take on write and read errors. Valid actions are: "ignore" (ignore the error and try to continue), "stop" (pause QEMU), "report" (report the error to the guest), "enospc" (pause QEMU only if the host disk is full; report the error to the guest otherwise). The default setting is werror=enospc and rerror=report.
